packages/devtools/lib/src/ui/ui_utils.dart

```diff
@@ -255,13 +255,23 @@ class RegisteredServiceExtensionButton {
 }
 
 bool tabDisabledByQuery(String key) {
+  final hideQueryParam = _getHideQueryParam();
+  return hideQueryParam?.split(',')?.contains(key) ?? false;
+}
+
+bool allTabsEnabledByQuery() {
+  final hideQueryParam = _getHideQueryParam();
+  return hideQueryParam?.split(',')?.contains('none') ?? false;
+}
+
+String _getHideQueryParam() {
```

Comment:
nit: slightly simpler and it would avoid a lot of re-parsing if you rewrote as 
Set<String> _pagesHidden;
Set<String> get pagesHidden {
  if (_pagesHidden != null) return _pagesHidden;
  // compute the set
}

Reply:
done.